adapters/huaweiads/huaweiads.go

```diff
@@ -1187,10 +1188,17 @@ func (a *adapter) extractAdmNative(adType int32, content *content, bidType openr
 				linkObject.ClickTrackers = append(linkObject.ClickTrackers, monitor.Url...)
 			}
 			if monitor.EventType == "imp" {
-				nativeResult.ImpTrackers = append(nativeResult.ImpTrackers, monitor.Url...)
+				for _, url := range monitor.Url {
+					var eventTracker nativeResponse.EventTracker
+					eventTracker.Event = native1.EventTypeImpression
+					eventTracker.Method = native1.EventTrackingMethodImage
+					eventTracker.URL = url
```

Comment:
iterator var `url` refers  to same address for value in each iterations. so prefer using index variable to assign value for `eventTracker.URL` 

```
for idx, url := range monitor.Url {
      ...
     eventTracker.URL = monitor.Url [idx]
     ...
}

```

Reply:
Done. But why tho? Is this some kind of best practices when it comes to golang?

Reply:
May I know what is the benefit of such implementation?

Reply:
loop variable `url`  gets overridden in every iteration with new value at same memory location. so it is advisible to use  index variable for assignment